sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala

```diff
@@ -350,6 +350,9 @@ object QueryTest {
     Row.fromSeq(row.toSeq.map {
       case null => null
       case d: java.math.BigDecimal => BigDecimal(d)
+      // Round up the values to avoid approximate error
+      case f : Float => BigDecimal(f).setScale(8, BigDecimal.RoundingMode.HALF_UP).toFloat
```

Comment:
I think is not correct, the error is relatively small, comparing to value of it. For example,  `1e15` could have a error as `1`